MacOS build broken (again) by bash

Commits 38b7f2125fa and 7e2aa46f64a on master via PRs #77985  and #77910 have broken MacOS 'make' again:

```
github.com/kubernetes/kubernetes/hack/lib/golang.sh: line 166: declare: -g: invalid option
declare: usage: declare [-afFirtx] [-p] [name[=value] ...]
!!! [0517 10:55:19] Call tree:
!!! [0517 10:55:19]  1: /Users/tpepper/src/github.com/kubernetes/kubernetes/hack/lib/init.sh:55 source(...)
!!! [0517 10:55:19]  2: build/../build/common.sh:37 source(...)
!!! [0517 10:55:19]  3: build/release.sh:29 source(...)
make: *** [quick-release] Error 1
```

where "MacOS" is a non-homebrew extended MacOS with old old bash.

This is also reported on other older platforms like CentOS.

Master needs fixed with a new commit, then the above PRs and #77362 and the new fix need cherry picked to 1.14 as 1.14.2 is currently not buildable as #76974 is cherry picked there.

:bash_fire"

--- a/hack/lib/golang.sh
+++ b/hack/lib/golang.sh
@@ -163,10 +163,10 @@ kube::golang::dedup() {
 # to readonly.
 # The configured vars will only contain platforms allowed by the
 # KUBE_SUPPORTED* vars at the top of this file.
-declare -a -g KUBE_SERVER_PLATFORMS
-declare -a -g KUBE_CLIENT_PLATFORMS
-declare -a -g KUBE_NODE_PLATFORMS
-declare -a -g KUBE_TEST_PLATFORMS
+declare -a KUBE_SERVER_PLATFORMS
+declare -a KUBE_CLIENT_PLATFORMS
+declare -a KUBE_NODE_PLATFORMS
+declare -a KUBE_TEST_PLATFORMS
 kube::golang::setup_platforms() {
   if [[ -n "${KUBE_BUILD_PLATFORMS:-}" ]]; then
     # KUBE_BUILD_PLATFORMS needs to be read into an array before the next
@@ -219,7 +219,7 @@ kube::golang::setup_platforms() {
       KUBE_CLIENT_PLATFORMS=(
         darwin/amd64
         linux/amd64
-      ) 
+      )
       readonly KUBE_CLIENT_PLATFORMS
     else
       KUBE_TEST_PLATFORMS=(linux/amd64)

